duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -175,299 +205,480 @@ namespace caf {
+
+        if( mcpart.StatusCode() == genie::EGHepStatus::kIStStableFinalState )
+        {
+          inter.prim.push_back(std::move(part));
+          inter.nprim++;
 
+          if( pdg == 2212 ) inter.nproton++;
+            else if( pdg == 2112 ) inter.nneutron++;
+            else if( pdg ==  211 ) inter.npip++;
+            else if( pdg == -211 ) inter.npim++;
+            else if( pdg ==  111 ) inter.npi0++;
+          }
+          else // kIStHadronInTheNucleus
+          {
+            inter.prefsi.push_back(std::move(part));
+            inter.nprefsi++;
         }
-      }
 
-      // Reweighting variables
+      }
 
-      // Consider
-      //systtools::ScrubUnityEventResponses(er);
+      truthBranch.nu.push_back(std::move(inter));
+    } // loop through MC truth i
 
-      sr.total_xsSyst_cv_wgt = 1;
+    truthBranch.nnu = mctruth.size();
+  }
 
-      for(auto &sp : fSystProviders ) {
-        std::unique_ptr<systtools::EventAndCVResponse> syst_resp = sp->GetEventVariationAndCVResponse(evt);
-        if( !syst_resp ) {
-          std::cout << "[ERROR]: Got nullptr systtools::EventResponse from provider "
-                    << sp->GetFullyQualifiedName();
-          abort();
-        }
+  //------------------------------------------------------------------------------
 
-        // The iteration order here is the same as how we filled SRGlobal, so
-        // no need to do any work to make sure they align.
-        //
-        // NB this will all go wrong if we ever support more than one MCTruth
-        // per event.
-        for(const std::vector<systtools::VarAndCVResponse>& resp: *syst_resp){
-          for(const systtools::VarAndCVResponse& it: resp){
-            // Need begin/end to convert double to float
-            sr.xsSyst_wgt.emplace_back(it.responses.begin(), it.responses.end());
-            sr.cvwgt.push_back(it.CV_response);
-            sr.total_xsSyst_cv_wgt *= it.CV_response;
+  void CAFMaker::FillRecoInfo(caf::SRCommonRecoBranch &recoBranch, const art::Event &evt) const {
+    SRInteractionBranch &ixn = recoBranch.ixn;
+
+    //Only filling with Pandora Reco for the moment
+    std::vector<SRInteraction> &pandora = ixn.pandora;
+    
+    lar_pandora::PFParticleVector particleVector;
+    lar_pandora::LArPandoraHelper::CollectPFParticles(evt, fPandoraNuVertexModuleLabel, particleVector);
+    lar_pandora::VertexVector vertexVector;
+    lar_pandora::PFParticlesToVertices particlesToVertices;
+    lar_pandora::LArPandoraHelper::CollectVertices(evt, fPandoraNuVertexModuleLabel, vertexVector, particlesToVertices);
+
+    for (unsigned int n = 0; n < particleVector.size(); ++n) {
+      const art::Ptr<recob::PFParticle> particle = particleVector.at(n);
+      if(particle->IsPrimary()){
+        SRInteraction reco;
+
+        //Retrieving the reco vertex
+        lar_pandora::PFParticlesToVertices::const_iterator vIter = particlesToVertices.find(particle);
```

Comment:
since you're only looking for likely, at most one vertex it might be quicker to use the `DUNEAnaUtils::DUNEAnaPFParticleUtils` getter functions to get the vertex for the neutrino particle. ACTUALLY, there's functions in here that will 1) tell you if you have a neutrino and 2) get it for you. So maybe that's cleaner as opposed to using the `->IsPrimary ` function, although I respect that this is motivated by the idea that you might have multiple neutrinos (which is true for the ND) and you might be aiming for the same format.

Reply:
For now I prefer to aim for the same format. Maybe in the future multiple primary particles will be required for some studies.